sycl/plugins/cuda/pi_cuda.cpp

```diff
@@ -4725,7 +4725,11 @@ pi_result cuda_piextUSMGetMemAllocInfo(pi_context context, const void *ptr,
                        PI_MEM_TYPE_HOST);
       }
       // should never get here
+#ifdef _MSC_VER
+      __assume(0);
+#else
       __builtin_unreachable();
+#endif
```

Comment:
this deserves to be in an utility function

Reply:
Do you have a location in mind for this utility function? As this function would not want to be relegated just to pi_cuda.cpp, it would not make sense to place this utility func in pi_cuda.h
Is there a location in dpc++ for these sorts of functions?

Reply:
I don't think we want plugins have dependency on the runtime, but we could create shared utilities for all plugins (many existing handling may be unified across plugins too). I think this deserves a separate PR and this can go in as is.